scipy/optimize/_constraints.py

```diff
@@ -165,11 +165,18 @@ def _input_validation(self):
 
     def __init__(self, A, lb=-np.inf, ub=np.inf, keep_feasible=False):
         if not issparse(A):
-            self.A = np.atleast_2d(A)
+            # In some cases, if the constraint is not valid, this emits a
+            # VisibleDeprecationWarning about ragged nested sequences
+            # before eventually causing an error. `scipy.optimize.milp` would
+            # prefer that this just error out immediately so it can handle it
+            # rather than concerning the user.
+            with catch_warnings():
+                simplefilter("error")
```

Comment:
Do the tests below cover this case?  I don't see any ragged arrays, but I suppose existing tests with a mix of `LinearConstraint` and arguments might trigger this?

Reply:
When the `constraints` input of `milp` has length 3, `milp` tries to convert it to a single `LinearConstraint`, the initializer of which accepts three arguments. If this conversion fails and an error is raised, `milp` knows that it is *not* supposed to be a single `LinearConstraint`, and it tries to interpret it another way. The problem we had is that the initializer of `LinearConstraint` was _not_ failing, so a sequence of the `LinearConstraint` objects or three tuples (each intended to become a `LinearConstraint`) were being collected into a single `LinearConstraint` object. Failures would occur later, because that is not a valid `LinearConstraint`.

The VisibleDeprecationWarning is triggered when three tuples (each of which is intended to be converted into a `LinearConstraint`) is passed as the `constraints` argument of `milp`. The first tuple - which has arrays of different shapes, because they are intended to be passed into the initializer as `A`, `lb` and `ub` separately - is treated as though it were all `A`. Without this `simplefilter`, `np.atleast_2d(A)` doesn't error out, but it emits the warning. Instead, I'm making it error out.